tests/integration/security/rbac/util/rbac_util.go

```diff
@@ -56,40 +62,21 @@ func (tc TestCase) Check() error {
 			err = resp.CheckOK()
 		}
 		if err != nil {
-			return fmt.Errorf("%s to %s:%s%s using %s: expected allow but was denied: %v",
-				req.From.Config().Service,
-				req.Options.Target.Config().Service,
-				req.Options.PortName,
-				req.Options.Path,
-				req.Options.Scheme,
-				err)
+			return getError(req, "allow with code 200", fmt.Sprintf("error: %v", err))
 		}
 	} else {
-		if err != nil && !strings.Contains(err.Error(), "EOF") {
-			return fmt.Errorf("connection error with %v", err)
-		}
-
 		if req.Options.PortName == "tcp" {
-			if err != nil && !strings.Contains(err.Error(), "EOF") {
-				return fmt.Errorf("%s to %s:%s%s using %s: expected deny with EOF error, actually %v",
-					req.From.Config().Service,
-					req.Options.Target.Config().Service,
-					req.Options.PortName,
-					req.Options.Path,
-					req.Options.Scheme,
-					err)
+			if err == nil || !strings.Contains(err.Error(), "EOF") {
+				return getError(req, "deny with EOF error", fmt.Sprintf("error: %v", err))
 			}
 		} else {
-			if !(len(resp) > 0 && resp[0].Code == tc.RejectionCode) {
-				return fmt.Errorf("%s to %s:%s%s using %s: expected deny, actually allow",
-					req.From.Config().Service,
-					req.Options.Target.Config().Service,
-					req.Options.PortName,
-					req.Options.Path,
-					req.Options.Scheme)
+			if err != nil {
+				return getError(req, "deny with code 403", fmt.Sprintf("error: %v", err))
```

Comment:
Where is 403 coming from? Check() is a public library function (the first letter of the function name is capitalized). The hard-coding here makes the Check() function difficult to understand (have to read the entire function implementation to understand this funciton) and to use as a libary function. Suggest getting rid of hard-coding and enabling callers to use it as a library function without reading the entire implementation.

Reply:
We can just rename the function back to `CheckRBACRequest`. It does not clear what `util.Check()` does. A user when using the `CheckRBACRequest` function should either expect the request to be denied or allowed. They don't need to read the implementation.

Reply:
I don't think it makes anything harder to understand. The `Check()` is in the RBAC package and its original name is `CheckRBACRequest`, we would only expect "403" for the denied request in this context. In what case you will specify a different code other than "403" in the caller for RBAC? Also if it really benefits to pass 403 in the caller, shouldn't we do the same thing for "200"?

Leaving only `ExpectedAllow` as the only field to specify the expected of the request is not only cleaner but also easier to understand. See similar rational in https://github.com/istio/istio/commit/f10b213e9464369aaef820008f86a4da14a80651#r33377100

@pitlv2109

Reply:
This class wasn't really changed from the original ... I just moved it around a bit, but the code should be more or less in tact. I did trim the name down to just `Check`, but that was due to the fact that it's already scoped to an `rbac` package ... having `RBAC` in the name here isn't useful, and is somewhat anti-idiomatic in Go.

Reply:
It's in the `util` package though (`rbac/util/rbac_util.go`). I think we should change the name back

For the other point, I agree with @yangminzhu. Like I said, the user should only expect if the request is denied or allowed, they should not need to know the detail of the implementation, i.e. http code.